packages/core/src/i18n.ts

```diff
@@ -232,13 +232,13 @@ export class I18n extends EventEmitter<Events> {
       this.emit("missing", { id, locale: this._locale })
     }
 
-    let translation = messageForId || message || id
-
-    if (process.env.NODE_ENV !== "production") {
-      translation = isString(translation)
-        ? compileMessage(translation)
-        : translation
-    }
+    // Bypass compilation for `messageForId`, but still compile `message || id`
+    // to avoid double compilation
+    // ref: https://github.com/lingui/js-lingui/issues/1901
```

Comment:
```js
// To avoid double compilation, skip compilation for `messageForId`, because message from catalog should be already compiled
// ref: https://github.com/lingui/js-lingui/issues/1901
```